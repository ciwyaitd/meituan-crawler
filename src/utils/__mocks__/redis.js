'use strict'

const redis = require('redis-mock')
const { promisify } = require('util')

const client = redis.createClient()

const getAsync = jest.fn(promisify(client.get).bind(client))
const setAsync = jest.fn(promisify(client.set).bind(client))
const keysAsync = jest.fn(promisify(client.keys).bind(client))
const hgetAsync = jest.fn(promisify(client.hget).bind(client))
const hsetAsync = jest.fn(promisify(client.hset).bind(client))
const hkeysAsync = jest.fn(promisify(client.hkeys).bind(client))
const hdelAsync = jest.fn(promisify(client.hdel).bind(client))
const hgetallAsync = jest.fn(promisify(client.hgetall).bind(client))
const delAsync = jest.fn(promisify(client.del).bind(client))
const saddAsync = promisify(client.sadd).bind(client)
const sismenberAsync = promisify(client.sismember).bind(client)

module.exports = {
  client,
  getAsync,
  setAsync,
  keysAsync,
  hgetAsync,
  hsetAsync,
  hkeysAsync,
  hdelAsync,
  hgetallAsync,
  delAsync,
  saddAsync,
  sismenberAsync
}
