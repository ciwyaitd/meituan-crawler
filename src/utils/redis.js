'use strict'

const redis = require('redis')
const config = require('config')
const { promisify } = require('util')
const logger = require('../utils/logger')

const redisConfig = {
  db: config.services.redis.db,
  host: config.services.redis.host,
  port: config.services.redis.port
}

if (config.services.redis.db === undefined || config.services.redis.db === null) {
  throw new Error('Please set environment variable COURIERS_DATA_REDIS_DATABASE_NUMBER')
}

if (config.services.redis.host === undefined || config.services.redis.host === null) {
  throw new Error('Please set environment variable COURIERS_DATA_REDIS_HOST')
}

if (config.services.redis.port === undefined || config.services.redis.port === null) {
  throw new Error('Please set environment variable COURIERS_DATA_REDIS_PORT')
}

let client = redis.createClient(redisConfig)

client.on('connect', () => {
  logger.info('[redis] Connect to redis.')
})

client.on('reconnecting', () => {
  logger.info('[redis] Reconnecting to redis.')
})

client.on('error', (err) => {
  logger.error(`[redis] Redis has some error: ${err.stack}`)
})

client.on('ready', () => {
  logger.info('[redis] Redis is ready.')
})

client.on('end', () => {
  logger.info('[redis] Redis is end.')
  client = redis.createClient(redisConfig)
})

const getAsync = promisify(client.get).bind(client)
const setAsync = promisify(client.set).bind(client)
const keysAsync = promisify(client.keys).bind(client)
const hgetAsync = promisify(client.hget).bind(client)
const hsetAsync = promisify(client.hset).bind(client)
const hkeysAsync = promisify(client.hkeys).bind(client)
const hgetallAsync = promisify(client.hgetall).bind(client)
const hdelAsync = promisify(client.hdel).bind(client)
const delAsync = promisify(client.del).bind(client)

module.exports = {
  client,
  getAsync,
  setAsync,
  keysAsync,
  hgetAsync,
  hsetAsync,
  hkeysAsync,
  hgetallAsync,
  hdelAsync,
  delAsync
}
