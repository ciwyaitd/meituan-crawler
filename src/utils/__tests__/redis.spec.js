'use strict'

describe('utils.redis', () => {
  beforeEach(() => jest.resetModules())
  it('should throw error when process.env.COURIERS_DATA_REDIS_DATABASE_NUMBER is undefined', () => {
    expect.assertions(2)
    try {
      require('../redis')
    } catch (e) {
      expect(e).toBeInstanceOf(Error)
      expect(e.message).toEqual('Please set environment variable COURIERS_DATA_REDIS_DATABASE_NUMBER')
    }
  })

  it('should throw error when process.env.COURIERS_DATA_REDIS_HOST is undefined', () => {
    process.env.COURIERS_DATA_REDIS_DATABASE_NUMBER = 0
    expect.assertions(2)
    try {
      require('../redis')
    } catch (e) {
      expect(e).toBeInstanceOf(Error)
      expect(e.message).toEqual('Please set environment variable COURIERS_DATA_REDIS_HOST')
    }
  })

  it('should throw error when process.env.COURIERS_DATA_REDIS_PORT is undefined', () => {
    process.env.COURIERS_DATA_REDIS_DATABASE_NUMBER = 0
    process.env.COURIERS_DATA_REDIS_HOST = '127.0.0.1'
    expect.assertions(2)
    try {
      require('../redis')
    } catch (e) {
      expect(e).toBeInstanceOf(Error)
      expect(e.message).toEqual('Please set environment variable COURIERS_DATA_REDIS_PORT')
    }
  })
})
