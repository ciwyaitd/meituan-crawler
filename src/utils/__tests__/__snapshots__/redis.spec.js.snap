// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils.redis should throw error when process.env.REDIS_DATABASE is undefined 1`] = `[Error: Please set environment variable REDIS_DATABASE]`;

exports[`utils.redis should throw error when process.env.REDIS_HOST is undefined 1`] = `[Error: Please set environment variable REDIS_HOST]`;

exports[`utils.redis should throw error when process.env.REDIS_PORT is undefined 1`] = `[Error: Please set environment variable REDIS_PORT]`;
